indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -181,7 +182,13 @@ private boolean createAndStartSupervisorInternal(SupervisorSpec spec, boolean pe
     supervisor.start(); // try starting the supervisor first so we don't persist a bad spec
 
     if (persistSpec) {
-      metadataSupervisorManager.insert(id, spec);
+      try {
+        metadataSupervisorManager.insert(id, spec);
+      } catch (Exception e) {
+        // Got exception while persisting the spec, stop the supervisor and propagate the exception
+        supervisor.stop(false);
```

Comment:
I checked the code again and stop gracefully set to `false` means running tasks are not killed. So, `false` flag is meant to handle cases like leadership changes. Here, we would like to just kill the tasks so doing what @cheddar suggested would be correct.